src/components/history-page/history-toolbar/history-toolbar.tsx

```diff
@@ -85,6 +83,31 @@ export const HistoryToolbar: React.FC<HistoryToolbarProps> = ({ onSettingsChange
     setState(prevState => ({ ...prevState, selectedTags: selected }))
   }
 
+  const refreshHistory = useCallback(() => {
+    refreshHistoryState().catch(
+      showErrorNotification(t('landing.history.error.getHistory.text'))
+    )
+  }, [t])
+
+  const onUploadAllToRemote = useCallback(() => {
+    if (!userExists) {
+      return
+    }
+    const entries = [...historyEntries]
```

Comment:
why do you copy this array here?

Reply:
Ahem, because of problems that have been resolved quite a while ago.
Is removed now.